packages/tools/src/utilities/dynamicVolume/generateImageFromTimeData.ts

```diff
@@ -25,17 +25,16 @@ function generateImageFromTimeData(
     throw new Error('Please provide two or more time points');
   }
 
-  // Gets scalar data for all time frames
-  const typedArrays = dynamicVolume.getScalarDataArrays();
+  const voxelManager = dynamicVolume.voxelManager;
 
-  const arrayLength = typedArrays[0].length;
+  const arrayLength = voxelManager.getScalarDataLength();
   const finalArray = new Float32Array(arrayLength);
 
   if (operation === Enums.DynamicOperatorType.SUM) {
```

Comment:
In general having an object of type { [key:Enums.DynamicOperatorType]: function } and then calling the object[operation] allows for generically adding other operation types by just defining the function, and that also isolates the code for each object into the particular function, which reduces cyclomatic complexity of the method.  I know it was like this before, so I'm not really expecting you to change it.